src/dns/server.rs

```diff
@@ -656,14 +664,33 @@ struct SystemForwarder {
 }
 
 impl SystemForwarder {
-    fn new() -> Result<Self, Error> {
+    fn new(custom_name_servers: Vec<SocketAddr>) -> Result<Self, Error> {
         // Get the resolver config from /etc/resolv.conf.
         let (cfg, opts) = read_system_conf()?;
 
         // Extract the parts.
         let domain = cfg.domain().cloned();
         let search_domains = cfg.search().to_vec();
-        let name_servers = cfg.name_servers().to_vec();
+        let mut name_servers = cfg.name_servers().to_vec();
+
+        for s in &custom_name_servers {
```

Comment:
Let's not merge. Instead, make the nameservers list an Option and if set, just overwrite the configuration we pass into the resolver.

Reply:
addressed, this code was removed